src/main/scala/is/hail/methods/PCRelate.scala

```diff
@@ -29,34 +33,173 @@ object PCRelate {
     def map[N](f: M => N): Result[N] = Result(f(phiHat), f(k0), f(k1), f(k2))
   }
 
-  val defaultMinKinship = Double.NegativeInfinity
+  val defaultMinKinship: Double = Double.NegativeInfinity
   val defaultStatisticSubset: StatisticSubset = PhiK2K0K1
+  val defaultStorageLevel: StorageLevel = StorageLevel.MEMORY_AND_DISK
+
+  private[this] def apply(
+    hc: HailContext,
+    blockedG: M,
+    columnKeys: Array[Annotation],
+    columnKeyType: Type,
+    pcs: DenseMatrix[Double],
+    maf: Double,
+    blockSize: Int,
+    minKinship: Double,
+    storageLevel: StorageLevel,
+    statistics: PCRelate.StatisticSubset): Table = {
+
+    val result = new PCRelate(maf, blockSize, statistics, storageLevel)(blockedG, pcs)
+
+    PCRelate.toTable(hc, result, columnKeys, columnKeyType, blockSize, minKinship, statistics)
+  }
+
+  def apply(
+    hc: HailContext,
+    blockedG: M,
+    columnKeys: Array[Annotation],
+    columnKeyType: Type,
+    pcaScores: Array[IndexedSeq[Double]],
+    maf: Double,
+    blockSize: Int,
+    minKinship: Double,
+    storageLevel: StorageLevel,
+    statistics: PCRelate.StatisticSubset): Table = {
+
+    val pcs = rowsToBDM(pcaScores)
+
+    apply(hc, blockedG, columnKeys, columnKeyType, pcs, maf, blockSize, minKinship, storageLevel, statistics)
+  }
+
+  def apply(
+    hc: HailContext,
+    blockedG: M,
+    scoresTable: Table,
+    maf: Double,
+    blockSize: Int,
+    minKinship: Double,
+    statistics: PCRelate.StatisticSubset): Table = {
+    
+    val scoresLocal = scoresTable.collect()
+    assert(scoresLocal.length == blockedG.nCols)
+
+    val keyQueriers = scoresTable.key.map(scoresTable.signature.query(_))        
```

Comment:
I think I proposed the wrong thing -- Instead, we can annotate the MT in Python and just pass in a field name for the PCs.